System/PathFinder.cpp

```diff
@@ -182,70 +171,57 @@ namespace RTE {
 		RTEAssert(g_SceneMan.GetScene(), "Scene doesn't exist or isn't loaded when recalculating PathFinder!");
 
 		// Update all the costs going out from each node
-		for (const std::vector<PathNode *> &nodeEntry : m_NodeGrid) {
-			for (PathNode *pathNode : nodeEntry) {
-				UpdateNodeCosts(pathNode);
-				// Should reset the updated flag since we're about to reset the pather
-				pathNode->IsUpdated = false;
+		std::for_each(
+			std::execution::par_unseq,
+			m_NodeGrid.begin(),
+			m_NodeGrid.end(),
+			[this](PathNode &node) {
+				UpdateNodeCosts(&node);
 			}
-		}
+		);
+
+		// Fix up the left-and-up connections
```

Comment:
I don't love that this can't happen in UpdateNodeCosts, and so needs to be redone in two places.
But I guess multithreading UpdateNodeCosts means this has to happen separately eh?